Simplify KeyringAddress interface

axelchalon opened the pull request:
Closes https://github.com/polkadot-js/ui/issues/133

For consistency, also had to change the interface of `@polkadot/keyring KeyringPair` ([PR](https://github.com/polkadot-js/common/pull/419)). Need to update the dependency once the other PR is merged.

axelchalon committed:
Simplify ui-keyring KeyringAddress

axelchalon committed:
Adapt to new @polkadot/keyring KeyPair interface

jacogr committed:
Merge branch 'master' into ac-getaddress

jacogr committed:
Pull in common 0.93.0-beta.4

jacogr reviewed (changes requested):
CI failures. Possibly due to this https://github.com/polkadot-js/ui/commit/c2e3d5fabe8c259344015c859610d2b12a12dac5 (common deps have been updated)

axelchalon committed:
Merge from master

axelchalon commented:
Modified `getAddress` after the merge. Made it so that unlike `KeyPair`, `getAddress` (more precisely the address prefix) is evaluated when calling `getAddress()`. This lets us return either `undefined` or a `KeyringAddress`, and if a KeyringAddress is returned, then we're sure it exists and we can access its properties (as such removed `isValid` as well), cf https://github.com/polkadot-js/ui/issues/133

This PR is a breaking change as well so I added a line in the changelog.

axelchalon committed:
KeyringAddress | undefined

axelchalon commented:
bump

on another note, `apps` still needs to be updated to use the new keyring, I can take care of this

jacogr commented:
I dunno how I missed you last changes. Last time I looked it had the merge issues and I swear it didn't pop up in my GH notifications. Thanks for the bump.

jacogr commented:
Yes, the affected ones are both apps and extension (although the latter _may_ be ok, I think - it doesn't do random lookups and actually don't think there are any throws around these - if you can't find an address, you have bigger issue there than catching an exception.).

jacogr reviewed (approved):
lgtm

Would appreciate the work in apps to pull this in. Stuff are (mostly) located in ui-app/util - there are helpers to get name, tags, etc. (There are probably cases where things are still done with throw/catch)

amaury1093 reviewed (approved):
lgtm
